### PR TITLE
Expanding the introduction and adding a FAQ doc

### PR DESCRIPTION
**Pull request by LalatenduMohanty**
Signed-off-by: Lalatendu Mohanty <lmohanty@redhat.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,7 +19,21 @@
   - [Documentation](#documentation)
 
 # OpenShift Hive
-API driven OpenShift cluster provisioning and management
+API driven OpenShift 4 cluster provisioning and management.
+
+Hive is an operator which runs as a service on top of OpenShift.
+The Hive service can be used to provision and manage OpenShift clusters.
+There is no limit on how many cluster Hive can provision or manage.
```
Comment (LalatenduMohanty) [inline README.md]: IMO one sentence per line for documentation is a good practice which helps in long term maintainability of the docs. Refer https://rhodesmill.org/brandon/2012/one-sentence-per-line/

**Comment by LalatenduMohanty** [comment]
Currently the we have a single line explaining the project i.e. `API driven OpenShift 4 cluster provisioning and management.` which does not give much context to folks who are new to the project. I do not think this PR does justice in explaining what is Hive does. The idea is to give enough context to folks so that they will feel confident about the basic nature of Hive.

File: README.md (hunk 1 above)
Comment (dgoodwin) [reply README.md]: I'm not sure this last sentence is a good idea, there is almost surely a limit, we just haven't tried to find out where it is yet.

File: README.md [hunk 2]
```diff
@@ -20,7 +19,21 @@
...
+
+* For provisioning OpenShift, Hive uses [installer](https://github.com/openshift/installer).
```
Comment (dgoodwin) [inline README.md]: "uses the OpenShift installer".

File: README.md (hunk 1 above)
Comment (dgoodwin) [reply README.md]: Could you fit in somewhere that hive's responsibilities are API driven provisioning, day 2 configuration management, reshaping, and deprovisioning.

File: README.md [hunk 3]
```diff
@@ -20,7 +19,21 @@
...
+* For management of hive-managed clusters, [SyncSet](./docs/syncset.md) object is used.
```
Comment (dgoodwin) [inline README.md]: "For day 2 configuration management, Hive provides the SyncSet CRD to deliver arbitrary Kubernetes configuration to the cluster."

File: docs/FAQs.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+- [Frequently Asked Questions](#frequently-asked-questions)
+  - [How is Hive different than installer?](#how-is-hive-different-than-installer)
+
+# Frequently Asked Questions
+
+## How is Hive different than installer?
```
Comment (dgoodwin) [inline docs/FAQs.md]: Should probably substitute installer with "OpenShift Installer" everywhere it's used.

File: README.md (hunk 1 above)
Comment (LalatenduMohanty) [reply README.md]: > I'm not sure this last sentence is a good idea, there is almost surely a limit, we just haven't tried to find out where it is yet.

Agree. I will move this sentence to FAQ.

File: README.md [hunk 5]
```diff
@@ -20,7 +19,20 @@
...
 
 # OpenShift Hive
-API driven OpenShift cluster provisioning and management
+API driven OpenShift 4 cluster provisioning and management.
+
+Hive is an operator which runs as a service on top of OpenShift.
+The Hive service can be used to provision and manage OpenShift clusters.
+
+* For provisioning OpenShift, Hive uses the [OpenShift installer](https://github.com/openshift/installer).
+* For `day 2` configuration management, Hive provides the [SyncSet](./docs/syncset.md) CRD to deliver arbitrary Kubernetes configuration to the cluster.
```
Comment (csrwng) [inline README.md]: s/[SyncSet]/[SyncSet and SelectorSyncSet CRDs]/ to deliver...

File: README.md [hunk 6]
```diff
@@ -20,7 +19,20 @@
...
+
+Supported cloud providers:
+* AWS
+
+In future Hive will support more cloud providers.
```
Comment (csrwng) [inline README.md]: s/In future/In the future/

File: docs/FAQs.md [hunk 7]
```diff
@@ -0,0 +1,12 @@
+- [Frequently Asked Questions](#frequently-asked-questions)
+  - [How is Hive different than OpenShift installer?](#how-is-hive-different-than-openshift-installer)
+
+# Frequently Asked Questions
+
+## How is Hive different than OpenShift installer?
+
+* OpenShift installer can be used to install/provision only one OpenShift cluster but Hive can be used for mutiple clusters.
```
[the remark this answers is not in the document]
Comment (csrwng) [reply docs/FAQs.md]: You also may want to mention that installation via Hive is done via an API whereas the installer requires a CLI

File: docs/FAQs.md [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+* There is no specific limit imposed in the code with respect to how many clusters Hive can provision or manage.
+How ever there would be surely a limit as the code can not scale infinitly.
```
[the remark this answers is not in the document]
Comment (csrwng) [reply docs/FAQs.md]: On second thought... not sure if this is relevant to how Hive and the installer are different.

File: docs/FAQs.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (LalatenduMohanty) [reply docs/FAQs.md]: Good point.

File: README.md [hunk 9]
```diff
@@ -20,7 +19,20 @@
   - [Documentation](#documentation)
 
 # OpenShift Hive
-API driven OpenShift cluster provisioning and management
+API driven OpenShift 4 cluster provisioning and management.
+
+Hive is an operator which runs as a service on top of OpenShift.
```
[the remark this answers is not in the document]
Comment (LalatenduMohanty) [reply README.md]: Right. Going to replace `OpenShift` with `Kubernetes/OpenShift`.

File: docs/FAQs.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (LalatenduMohanty) [reply docs/FAQs.md]: Going to remove this point.

File: docs/FAQs.md [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+* OpenShift installer does not manage OpenShift cluster.
+* Hive can be only used for public clouds.
```
[the remark this answers is not in the document]
Comment (LalatenduMohanty) [reply docs/FAQs.md]: > A use case for Hive would be the adoption of clusters installed via the UPI installer that you still want to bring under management.

Good point, removed the point.

**Comment by csrwng** [comment]
/lgtm cancel
still have a couple of nits

File: README.md [hunk 11]
```diff
@@ -20,7 +19,20 @@
...
+
+* For provisioning OpenShift, Hive uses the [OpenShift installer](https://github.com/openshift/installer).
+* For `day 2` configuration management, Hive provides the [SyncSet and SelectorSyncSet CRDs](./docs/syncset.md) to deliver arbitrary Kubernetes configuration to the cluster.
+
+Supported cloud providers:
+* AWS
+
+In the future Hive will support more cloud providers.
+
+For more information, refer to the [documention](#documentation).
```
Comment (csrwng) [inline README.md]: s/documention/documentation/

File: docs/FAQs.md [hunk 12]
```diff
@@ -0,0 +1,10 @@
+- [Frequently Asked Questions](#frequently-asked-questions)
+  - [How is Hive different than the OpenShift installer?](#how-is-hive-different-than-the-openshift-installer)
+
+# Frequently Asked Questions
+
+## How is Hive different than the OpenShift installer?
+
+* The OpenShift installer can be used to install/provision only one OpenShift cluster but Hive can be used for mutiple clusters.
+* Hive uses API to do provisioning/installation whereas the OpenShift installer requires a CLI.
```
Comment (csrwng) [inline docs/FAQs.md]: /uses API/exposes an API/

**Pull request merged**